posts/intake_esm_dask.md

```diff
@@ -0,0 +1,197 @@
+---
+author: Max Grover
+date: 2021-4-9
+tags: dask, intake, cesm
+---
+
+# An Example of Using Intake-ESM
+
+This past week, NCAR CISL updated the Casper Node to use PBS instead of Slurm for scheduling jobs. This led a post in which an example of spinning up dask clusters on the new configuration. This was also an opportunity to dig into dask, and try applying it to a sample task, specifically looking at Krill using CESM-LE data, using notebooks included in Matt Long's [krill-cesm-le repository](https://github.com/matt-long/krill-cesm-le), modified by Kristen Krumhardt.
+
+## Setting up the imports/dask
+
+```python
+%matplotlib inline
+import os
+import shutil
+
+from glob import glob
+
+import cftime
+
+import numpy as np
+import xarray as xr
+
+import matplotlib.pyplot as plt
+import matplotlib.colors as colors
+
+import cartopy.crs as ccrs
+from cartopy.util import add_cyclic_point
+
+import intake
+import pop_tools
+import esmlab
+import util
+
+import warnings
+warnings.filterwarnings('ignore')
+```
+
+### Spin up Dask Cluster
+
+Here, we spin up our dask cluster. At first, running this notebook resulted in a `killed worker` error. After further expection, we noticed that additional resources would be needed to read in the notebook since the data are so large (on the order of ~1-2 TB). Increasing the individual worker to a higher amount (ex. 256 GB) solved the issue. Scale up to as many workers as you think are neccessary for the calculation (this may take some trial and error).
+
```

Comment:
Or chunk accoring to your analysis. This only requires the top 100m so chunk in `z`